core/src/main/scala/com/salesforce/op/OpWorkflowModelReader.scala

```diff
@@ -234,20 +235,31 @@ class OpWorkflowModelReader(val workflowOpt: Option[OpWorkflow], val asSpark: Bo
 private object WorkflowFileReader {
 
   def loadFile(pathString: String)(implicit conf: Configuration): String = {
-    val path = new Path(pathString)
-    val fs = path.getFileSystem(conf)
-    val allFiles = fs.listFiles(path, false)
-    var partPath: Option[Path] = None
-    while (allFiles.hasNext) {
-      val p = allFiles.next().getPath
-      if (p.getName.startsWith("part-00000")) {
-        partPath = Option(p)
-      }
+    Try {
+      readAsString( new Path(pathString, "part-00000.gz") )
+    } recoverWith { case suppressed1 =>
+      Try {
+        readAsString( new Path(pathString, "part-00000") )
+      } recoverWith { case suppressed2 => suppressed2.addSuppressed(suppressed1)
+          Try {
+            readAsString( new Path(pathString) )
+          } recoverWith {
+            case th => th.addSuppressed(suppressed2)
+              Failure(th)
+          }
+        }
+      } match {
+      case Failure(e) => throw new RuntimeException(s"Failed to load workflow because of $e")
```

Comment:
Let us keep `e` as the cause exception as well, i.e. provided as the second parameter to the RTE constructor.